compiler/src/iree/compiler/GlobalOptimization/test/flow_hoist_into_globals.mlir

```diff
@@ -78,3 +78,15 @@ module @hoist_sub_byte_tensor_transitive {
 }
 // We do not need to cast for transitive sub-byte values.
 // CHECK-NOT: flow.tensor.bitcast
+
+// -----
+
+// CHECK-LABEL: @hoist_sub_byte_aligned_scalar_transitive
+// CHECK: util.global
+module @hoist_sub_byte_aligned_scalar_transitive {
+ func.func @main() -> i4 {
+    %c1_i4 = arith.constant 1 : i4
+    %0 = "iree_unregistered.const_expr"(%c1_i4) : (i4) -> i4
+    return %0 : i4
+  }
+}
```

Comment:
These checks should probably check the type of the global as well. Now that we have the bitcasting trick for global hoisting, I think it's good to be clear about this in the test cases.

nit: new line at eof